lib/core/CStatistics.cc

```diff
@@ -196,5 +204,21 @@ std::ostream& operator<<(std::ostream& o, const CStatistics& /*stats*/) {
     return o;
 }
 
+CStatistics::CStatsCache::CStatsCache() {
+}
+
+void CStatistics::CStatsCache::populate(const CStatistics::TStatArray& statArray) {
```

Comment:
It would be safer to have the constructor do the population.  If this `populate()` method is accidentally called more than once for the same object then it just appends, resulting in twice as many statistics as expected.